lending/loan_management/doctype/loan_balance_adjustment/loan_balance_adjustment.py

```diff
@@ -54,11 +55,15 @@ def validate(self):
 
 	def on_submit(self):
 		self.set_status_and_amounts()
-		self.make_gl_entries()
+		if loan_accounting_enabled(self.company):
```

Comment:
Getting a different idea here.
Make a LoanController class like AccountsController and StockController, and let all doctypes inherit the LoanController (this will be inherited from the accounts controller), then this check can only be done at the LoanController level instead of checking in each doctype.